packages/sanes-chrome-extension/src/extension/messages.ts

```diff
@@ -81,3 +81,27 @@ export async function sendGetPersonaMessage(): Promise<GetPersonaResponse> {
     });
   });
 }
+
+/**
+ * String values will be used in the JSON message objects
+ */
+export enum MessageToForegroundAction {
+  TransactionsChanges = 'transaction_changed',
+}
+
+/**
+ * A message either from background script to foreground
+ */
+export interface MessageToForeground {
+  readonly type: 'message_to_foreground';
+  readonly action: MessageToForegroundAction;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  readonly data?: any;
+}
+
+export function isMessageToForeground(data: unknown): data is MessageToForeground {
```

Comment:
This function should be unit tested for understanding when this could be a problem. If this is trivial, then just a /** doc is enough.

Reply:
This is a trivial type checker. You can put anything into `data` and it telly you if that is data of type `MessageToForeground`. We need this since messages come as type  `any`.